langserver/handlers/did_save.go

```diff
@@ -0,0 +1,59 @@
+package handlers
+
+import (
+	"context"
+
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/langserver/diagnostics"
+)
+
+func (h *logHandler) TextDocumentDidSave(ctx context.Context, params lsp.DidSaveTextDocumentParams) error {
+	fs, err := lsctx.DocumentStorage(ctx)
```

Comment:
To comply with the protocol I believe we should be gating the execution of this handler on the [relevant client capability](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_didSave). Technically this would just uncover buggy clients which attempt to call `didSave` without actually announcing support for it, but we should still do it IMO.

Returning a custom error with a message suggesting a bug on the client side is probably best.